### ISSUE TITLE
`NeighborLoader` not respecting `input_nodes`

### ISSUE THREAD

**Issue by feelingsonice** [pyg-team/pytorch_geometric#10144]
### 🐛 Describe the bug

```python
import torch
from torch_geometric.data import Data
from torch_geometric.loader import NeighborLoader
from torch_geometric.sampler import SamplerOutput

torch.manual_seed(12345)

# Graph: 0→1→2→3→4→5→6→0
edge_index = torch.tensor([
    [0, 1, 2, 3, 4, 5, 6],  # Source nodes
    [1, 2, 3, 4, 5, 6, 0],  # Target nodes
], dtype=torch.long)

# Simple node features: x[i] = i
x = torch.arange(7, dtype=torch.float).view(-1, 1)
data = Data(x=x, edge_index=edge_index)
input_nodes=torch.tensor([1, 3, 5])

print("=== Entire Graph ===")
print("Nodes:\t\t", data.num_nodes)         # 7
print("Edges:\t\t", data.num_edges)         # 7 (in a ring)
print("input_nodes:\t", input_nodes)        # choose [1, 3, 5] as seeds

loader = NeighborLoader(
    data,
    num_neighbors=[1, 1],                   # Up to 1 neighbors each hop, for 1 hops
    batch_size=2,                                 # Pick 2 seed nodes at a time
    input_nodes=input_nodes,           # <---------- NOT RESPECTED
    shuffle=False,
)

for i, batch in enumerate(loader):
    print(f"\n=== Mini-batch {i} ===")
    print("Current seed nodes (input_id):", batch.input_id.tolist())
```

Expected output:
```
=== Entire Graph ===
Nodes:		 7
Edges:		 7
input_nodes:	 tensor([1, 3, 5])

=== Mini-batch 0 ===
Current seed nodes (input_id): [1, 3]

=== Mini-batch 1 ===
Current seed nodes (input_id): [5]
```

But got:
```
=== Entire Graph ===
Nodes:		 7
Edges:		 7
input_nodes:	 tensor([1, 3, 5])

=== Mini-batch 0 ===
Current seed nodes (input_id): [0, 1]

=== Mini-batch 1 ===
Current seed nodes (input_id): [2]
```

Am I missing something here? My understanding is that `input_nodes` defines the initial seed nodes?

### Versions

Collecting environment information...
PyTorch version: 2.6.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 15.3.2 (arm64)
GCC version: Could not collect
Clang version: 16.0.0 (clang-1600.0.26.6)
CMake version: version 3.30.2
Libc version: N/A

Python version: 3.12.9 | packaged by conda-forge | (main, Mar  4 2025, 22:44:42) [Clang 18.1.8 ] (64-bit runtime)
Python platform: macOS-15.3.2-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M2 Max

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] optree==0.14.1
[pip3] torch==2.6.0
[pip3] torch-geometric==2.6.1
[pip3] torchvision==0.21.0
[pip3] torchvision-extra-decoders==0.0.2
[conda] libtorch                  2.6.0           cpu_generic_h6100933_3    conda-forge
[conda] nomkl                     1.0                  h5ca1d4c_0    conda-forge
[conda] numpy                     1.26.4          py312h8442bc7_0    conda-forge
[conda] optree                    0.14.1          py312hb23fbb9_1    conda-forge
[conda] pytorch                   2.6.0           cpu_generic_py312_heeb16a7_3    conda-forge
[conda] torch-geometric           2.6.1              pyhd8ed1ab_0    conda-forge
[conda] torchvision               0.21.0          cpu_py312_h6393126_1    conda-forge
[conda] torchvision-extra-decoders 0.0.2           py312h2e63e4f_2    conda-forge

**Comment by xnuohz** [comment]
try `input_nodes = [0, 1, 0, 1, 0, 1, 0]`
